api/server/methods.go

```diff
@@ -33,11 +42,54 @@ func (s *Server) ForceDBUpdate() {
 //   - inbox_max_count: assertions that have a specified value for InboxMaxCount
 //   - from_block_number: items that were created since a specific block number. Defaults to latest confirmed assertion
 //   - to_block_number: caps the response to assertions up to and including a block number
+//   - challenged: fetch only assertions that have been challenged
+//   - force_update: refetch the updatable fields of each item in the response
 //
 // response:
 // - []*JsonAssertion
-func (s *Server) ListAssertions() {
-
+func (s *Server) ListAssertions(r *http.Request, w http.ResponseWriter) {
+	opts := make([]db.AssertionOption, 0)
+	query := r.URL.Query()
+	if val, ok := query["limit"]; ok && len(val) > 0 {
+		if v, err := strconv.Atoi(val[0]); err == nil {
+			opts = append(opts, db.WithAssertionLimit(v))
+		}
+	}
+	if val, ok := query["offset"]; ok && len(val) > 0 {
+		if v, err := strconv.Atoi(val[0]); err == nil {
+			opts = append(opts, db.WithAssertionOffset(v))
+		}
+	}
+	if val, ok := query["inbox_max_count"]; ok && len(val) > 0 {
+		opts = append(opts, db.WithInboxMaxCount(strings.Join(val, "")))
+	}
+	if val, ok := query["from_block_number"]; ok && len(val) > 0 {
+		if v, err := strconv.ParseUint(val[0], 10, 64); err == nil {
+			opts = append(opts, db.FromAssertionCreationBlock(v))
+		}
+	}
+	if val, ok := query["to_block_number"]; ok && len(val) > 0 {
+		if v, err := strconv.ParseUint(val[0], 10, 64); err == nil {
+			opts = append(opts, db.ToAssertionCreationBlock(v))
+		}
+	}
+	if _, ok := query["challenged"]; ok {
+		opts = append(opts, db.WithChallenge())
+	}
+	if _, ok := query["force_update"]; ok {
+		opts = append(opts, db.WithAssertionForceUpdate())
+	}
+	assertions, err := s.backend.GetAssertions(r.Context(), opts...)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Could not get assertions from backend: %v", err), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", contentType)
+	if err = json.NewEncoder(w).Encode(assertions); err != nil {
```

Comment:
I think once the response body starts writing, the header should already be set, otherwise, Go's http package automatically sets the status to 200 OK: 
https://www.reddit.com/r/golang/comments/13sky9d/how_goes_go_know_which_http_status_code_to_send/

Reply:
Nice catch, done.